### PR TITLE
Create GUIHumanPlayer to receive new GUI player move.

### PR DESCRIPTION
**Pull request by sarahabimay**
@maikon
@jsuchy
GUIHumanPlayer is a type of Player.
GUIHumanPlayer receives the result of Board button click and then when
Game asks it's Player for it's next move then this move is provided.

### CODE REVIEW

File: test/jttt/Core/GameTest.java [hunk 1]
```diff
@@ -58,19 +59,7 @@ public void askedForFirstPlayer() {
...
-
-    @Test
-    public void askedForNextPlayer() {
-        defaultGame.playMove(1);
-        Player playerB = defaultGame.getNextPlayer();
-        assertEquals(Mark.O, playerB.getMark());
-    }
-
-    @Test
+    @Ignore
```
Comment (jsuchy) [inline test/jttt/Core/GameTest.java:62]: ignored tests shouldn't be committed / pushed.

File: test/jttt/Core/Fakes/FakeCommandLineUI.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
     private Game game;
     private List<Integer> dummyInputs = new ArrayList<>();
     private int playerType = 1;
-    private boolean playAgain = false;
+    private final int doNotPlayAgainOption = 2;
```
Comment (jsuchy) [inline test/jttt/Core/Fakes/FakeCommandLineUI.java:19]: is this a magic number? who else knows the meaning of 2?

File: GUI/test/javafxgui/EventHandlerTest.java [hunk 3]
```diff
@@ -10,16 +10,18 @@
 public class EventHandlerTest {
 
     private GUIViewSpy guiViewSpy;
+    private NewPlayerMoveEventHandler newMoveEventHandler;
 
     @Before
     public void setUp() {
         guiViewSpy = new GUIViewSpy(new Scene(new StackPane(), 700, 600), new BoardDisplay(), new EventRegister());
```
Comment (jsuchy) [inline GUI/test/javafxgui/EventHandlerTest.java:17]: I've seen the 700, 600 values together at least twice. Is that duplication?

File: test/jttt/Core/Players/PlayerFactoryTest.java [hunk 4]
```diff
@@ -22,14 +23,15 @@ public void setUp() throws Exception {
...
-        Assert.assertEquals(false, playerFactory.validPlayerTypes(0));
+        Assert.assertEquals(false, playerFactory.isValidGameType(0));
     }
 
     @Test
     public void validGameTypeChoice() {
-        Assert.assertEquals(true, playerFactory.validPlayerTypes(1));
-        Assert.assertEquals(true, playerFactory.validPlayerTypes(2));
-        Assert.assertEquals(true, playerFactory.validPlayerTypes(3));
+        Assert.assertEquals(true, playerFactory.isValidGameType(1));
```
Comment (jsuchy) [inline test/jttt/Core/Players/PlayerFactoryTest.java:31]: Similar to above, these seem like magic numbers. Can we give them a name, or perhaps use an enum?

File: test/jttt/Core/UI/CommandLineUITest.java [hunk 5]
```diff
@@ -17,7 +17,8 @@
 
 public class CommandLineUITest {
 
-    private final int CVH_GAME_TYPE = 3;
+    private final int CVH_GAME_TYPE_OPTION = 3;
```
Comment (jsuchy) [inline test/jttt/Core/UI/CommandLineUITest.java:20]: And I think this is where you created a couple constants for these magic numbers, but they're only available from the test?

File: test/jttt/Core/Fakes/FakeCommandLineUI.java [hunk 6]
```diff
@@ -165,4 +166,22 @@ private boolean isEndOfRow(int index, Board board) {
     private int calculateDimension(Board board) {
         return (int) Math.sqrt(board.boardSize());
     }
+
+    private boolean userWantsToPlay(int choice) {
+        return BinaryChoice.YES.equalsChoice(choice);
+    }
+    private enum BinaryChoice {
+        YES(1),
```
Comment (sarahabimay) [inline test/jttt/Core/Fakes/FakeCommandLineUI.java:174]: Make this enum class a real class not inner as also in CommandLineUI

**Pull request merged**